pkg/service/service.go

```diff
@@ -544,11 +544,19 @@ func (s *Service) createSVCInfoIfNotExist(p *lb.SVC) (*svcInfo, bool, bool,
 		// as the service clusterIP type. In such cases, if a Local redirect service
 		// exists, we shouldn't override it with clusterIP type (e.g., k8s event/sync, etc).
 		if svc.svcType == lb.SVCTypeLocalRedirect && p.Type == lb.SVCTypeClusterIP {
-			err := fmt.Errorf("Local redirect service exists for "+
-				"frontend %v, skip update for svc %s", p.Frontend, p.Name)
+			err := fmt.Errorf("local-redirect service exists for "+
+				"frontend %v, skip update for svc %v", p.Frontend, p.Name)
 			return svc, !found, prevSessionAffinity, prevLoadBalancerSourceRanges, err
 
 		}
+		// Local-redirect service can only override clusterIP service type or itself.
+		if p.Type == lb.SVCTypeLocalRedirect &&
+			(svc.svcType != lb.SVCTypeClusterIP && svc.svcType != lb.SVCTypeLocalRedirect) {
+			err := fmt.Errorf("skip local-redirect service for "+
+				"frontend %v as it overlaps with svc %v of type %v",
+				p.Frontend, svc.svcName, svc.svcType)
```

Comment:
Nit: the error msg should state what's wrong. In this case, the fact that the service overlapped with another, is the problem. Mentioning the action to be taken, `skipping`, is irrelevant in an error. That should be reserved for a log msg.

```suggestion
			err := fmt.Errorf("local-redirect service overlaps with %v of type %v", svc.svcName, svc.svcType)
```

Something like that.

Reply:
If you expand the commit description for the first commit, you will see the summary of changes. The changes are simple enough that I don't think it's necessary to further split them out.

Reply:
In general, we tend to make renames and other cosmetic changes as separate commits, as Joe pointed out. In this case, the changes are straightforward and small enough. However, the commit title as "Address leftover nits" is not helpful when reviewing the git log in the future. We should express what changed in the title and then why in the body. Could we change it, so that we have a git log with digestible pieces of context into what changed?

Reply:
done

Reply:
> Nit: the error msg should state what's wrong. In this case, the fact that the service overlapped with another, is the problem. Mentioning the action to be taken, `skipping`, is irrelevant in an error. That should be reserved for a log msg.
> 
> Something like that.

TMS. I'll probably also add structured logging here.

Reply:
I thought about this a bit more. The cilium-agent service module should ideally notify the caller (LRP manager/CLI) that it's going to skip the service update because of the overlap so the action _is_ relevant here, and should be included in the error message? 
@brb Any comments?

Reply:
In general, if the caller cares specifically about the error, then defining the error as a sentinel type [1] is the pattern, so that the caller can do `errors.Is(err, ErrOverlappingService)`, for example.

[1]:

```golang
type ErrOverlappingService struct {
    svc string
    type string
}
func (e ErrOverlappingService) Error() string { return fmt.Sprintf("....%s ... %v", e.svc, e.type) }
```

This is just a suggestion. I'm not too familiar with this code to suggest whether we should do this, but if it is a common enough error, then the caller can do the `errors.Is` and then log / alert the user that _it_ will skip the update, etc.

If that doesn't quite make sense in this case, then feel free to ignore. But just wanted to share how this pattern can be useful in keeping error msgs short and to the point, and keeping the responsibilities of the caller / callee clear. :smile:

Reply:
Sure. In this case, the action to skip a service update is taken by the service module so it makes sense to pass that information to the caller.